tools/repository_bumper.sh

```diff
@@ -0,0 +1,453 @@
+    local setup_file="${DIR_API}/setup.py"
+    local spec_file="${DIR_API}/api/spec/spec.yaml"
+
+    [[ -z "$new_version" && -z "$new_stage" ]] && return
+
+    if [[ -n "$new_version" ]]; then
+        # setup.py: version='X.Y.Z',
+        sed -i -E "s/^([[:space:]]+version=')[^']+(',)$/\1${new_version}\2/" "$setup_file"
+
+        # spec.yaml: version: 'X.Y.Z'
+        sed -i -E "s/^([[:space:]]+version:[[:space:]]+')([0-9]+\.[0-9]+\.[0-9]+)(')$/\1${new_version}\3/" "$spec_file"
+
+        # /vX.Y.Z/ => path to API
+        sed -i -E "s|(\/v)[0-9]+\.[0-9]+\.[0-9]+(/)|\1${new_version}\2|g" "$spec_file"
+
+        # com/X.Y/ => only the first two digits
+        VERSION_SHORT=$(echo "$new_version" | awk -F. '{print $1"."$2}')
+        sed -i -E "s|(com/)[0-9]+\.[0-9]+(/)|\1${VERSION_SHORT}\2|g" "$spec_file"
+        log_action "Updated version to '${new_version}' in: $setup_file and $spec_file"
+    fi
+
+    if [[ -n "$new_stage" ]]; then
+        # x-revision: 'alpha0' → x-revision: 'beta0'
+        sed -i -E "s/^([[:space:]]+x-revision:[[:space:]]+')([^']+)(')$/\1${new_stage}\3/" "$spec_file"
+        log_action "Updated revision to '${new_stage}' in: $spec_file"
+    fi
+}
+
+update_file_packages() {
+    local final_version="$1"
+    local final_stage="$2"
+    local new_date="$3"
+
+    # Split the version into major, minor and patch
+    IFS='.' read -r major minor patch <<< "$final_version"
+
+    # Format the date
+    formatted_date=$(date -d "$new_date" +"%a, %d %b %Y 00:00:00 +0000")
+    spec_date=$(date -d "$new_date" +"%a %b %d %Y")
+
+    # Update .spec files
+    for spec_file in $(find "$DIR_PACKAGE" -type f -name "*.spec"); do
+        version_line="* .* - ${final_version}"
+        existing_line=$(grep -E "^\\* .+ - ${final_version}$" "$spec_file")
+
+        if [ -n "$existing_line" ]; then
+            sed -i -E "s|^\* .+ - ${final_version}$|* ${spec_date} support <info@wazuh.com> - ${final_version}|" "$spec_file"
+            log_action "Updated changelog date for version ${final_version} in: $spec_file"
+        else
+            sed -i -E "/^%changelog\s*$/a * ${spec_date} support <info@wazuh.com> - ${final_version}\n- More info: https://documentation.wazuh.com/current/release-notes/release-${final_version//./-}.html" "$spec_file"
+            log_action "Prepended changelog entry for version ${final_version} in: $spec_file"
+        fi
+    done
+
+    # Update changelog files (prepend entry)
+    for changelog_file in $(find "$DIR_PACKAGE" -type f -name "changelog"); do
+        INSTALL_TYPE=$(basename "$(dirname "$(dirname "$changelog_file")")")
+
+        changelog_entry="$(
+cat <<EOF
+${INSTALL_TYPE} (${final_version}-RELEASE) stable; urgency=low
+
+* More info: https://documentation.wazuh.com/current/release-notes/release-${final_version//./-}.html
+
+-- Wazuh, Inc <info@wazuh.com>  ${formatted_date}
+
+EOF
+)"
+
+        version_pattern="^${INSTALL_TYPE} \(${final_version//./\\.}-RELEASE\) stable; urgency=low"
```

Comment:
This script works, but check the warnings:

```bash
╰─# bash tools/repository_bumper.sh --version 4.13.0 --stage alpha0 --date 2025-04-21
awk: warning: escape sequence `\(' treated as plain `('
awk: warning: escape sequence `\.' treated as plain `.'
awk: warning: escape sequence `\)' treated as plain `)'
awk: warning: escape sequence `\(' treated as plain `('
awk: warning: escape sequence `\.' treated as plain `.'
awk: warning: escape sequence `\)' treated as plain `)'
```

```suggestion
        version_pattern="^${INSTALL_TYPE} (${final_version//./\\.}-RELEASE) stable; urgency=low"
```